web/src/lib/components/elements/slider.svelte

```diff
@@ -1,26 +1,35 @@
 <script lang="ts">
   import { createEventDispatcher } from 'svelte';
 
+  /**
+   * Unique identifier for the checkbox element, used to associate labels with the input element.
+   */
+  export let id: string;
   export let checked = false;
   export let disabled = false;
 
+  let isFocused = false;
+
   const dispatch = createEventDispatcher<{ toggle: boolean }>();
   const onToggle = (event: Event) => dispatch('toggle', (event.target as HTMLInputElement).checked);
 </script>
 
 <label class="relative inline-block h-[10px] w-[36px] flex-none">
   <input
+    {id}
     class="disabled::cursor-not-allowed h-0 w-0 opacity-0"
     type="checkbox"
     bind:checked
     on:click={onToggle}
+    on:focus={() => (isFocused = true)}
+    on:blur={() => (isFocused = false)}
     {disabled}
   />
 
   {#if disabled}
-    <span class="slider slider-disabled cursor-not-allowed" />
+    <span class="slider slider-disabled cursor-not-allowed" class:focused={isFocused} />
   {:else}
-    <span class="slider slider-enabled cursor-pointer" />
+    <span class="slider slider-enabled cursor-pointer" class:focused={isFocused} />
```

Comment:
It's probably better to only show focus styling with `.slider:focus-visible` instead of regular focus. That also prevents the outline from flickering while clicking the slider. Also just using Tailwind or CSS would be easier instead of detecting focus with JS:

```svelte
<input class="peer" />
<span class="peer-focus-visible:outline" />
```